cmd/clusterctl/examples/openstack/generate-yaml.sh

```diff
@@ -197,91 +157,31 @@ tenant-id=\"$PROJECT_ID\"
-  MASTER_USER_DATA=$(echo "$MASTER_USER_DATA_PLAIN" \
-    | sed -e "s#\$OPENSTACK_CLOUD_PROVIDER_CONF#$OPENSTACK_CLOUD_PROVIDER_CONF#" \
-    | base64)
-  WORKER_USER_DATA=$(echo "$WORKER_USER_DATA_PLAIN" \
-    | sed -e "s#\$OPENSTACK_CLOUD_PROVIDER_CONF#$OPENSTACK_CLOUD_PROVIDER_CONF#" \
-    | base64)
 else
   echo "Unrecognized OS : $OS"
   exit 1
 fi
 
-# write config file to PROVIDERCOMPONENT_GENERATED_FILE
-for file in `ls "${PROVIDER_CRD_DIR}"`
-do
-    cat "${PROVIDER_CRD_DIR}/${file}" > "$PROVIDERCOMPONENT_GENERATED_FILE"
-    echo "---" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-done
-for file in `ls "${PROVIDER_RBAC_DIR}"`
-do
-    cat "${PROVIDER_RBAC_DIR}/${file}" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-    echo "---" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-done
-for file in `ls "${PROVIDER_MANAGER_DIR}"`
-do
-    sed "s/{OS_CLOUD}/$OS_CLOUD/g" "${PROVIDER_MANAGER_DIR}/${file}" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-    echo "---" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-done
-for file in `ls "${CLUSTER_MANAGER_DIR}"`
-do
-    cat "${CLUSTER_MANAGER_DIR}/${file}" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-    echo "---" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-done
-for file in `ls "${CLUSTER_CRD_DIR}"`
-do
-    cat "${CLUSTER_CRD_DIR}/${file}" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-    echo "---" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-done
-for file in `ls "${CLUSTER_RBAC_DIR}"`
-do
-    cat "${CLUSTER_RBAC_DIR}/${file}" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-    echo "---" >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-done
-
-cat "$PROVIDERCOMPONENT_TEMPLATE_FILE" \
-  | sed -e "s#\$OPENSTACK_CLOUD_CONFIG#$OPENSTACK_CLOUD_CONFIG#" \
-  | sed -e "s#\$MASTER_USER_DATA#$MASTER_USER_DATA#" \
-  | sed -e "s#\$WORKER_USER_DATA#$WORKER_USER_DATA#" \
-  >> "$PROVIDERCOMPONENT_GENERATED_FILE"
-
-if [[ "$OS" =~ "Linux" ]]; then
-  sed -i "s#image: controller:latest#image: gcr.io/k8s-cluster-api/cluster-api-controller:$CONTROLLER_IMAGE_VERSION#" "$PROVIDERCOMPONENT_GENERATED_FILE"
-elif [[ "$OS" =~ "Darwin" ]]; then
-  sed -i '' -e "s#image: controller:latest#image: gcr.io/k8s-cluster-api/cluster-api-controller:$CONTROLLER_IMAGE_VERSION#" "$PROVIDERCOMPONENT_GENERATED_FILE"
-else
-  echo "Unrecognized OS : $OS"
-  exit 1
-fi
-
-cat "$MACHINE_TEMPLATE_FILE" \
-  > "$MACHINE_GENERATED_FILE"
-
-cat "$CLUSTER_TEMPLATE_FILE" \
-  > "$CLUSTER_GENERATED_FILE"
-
-
-echo "Done generating $PROVIDERCOMPONENT_GENERATED_FILE $MACHINE_GENERATED_FILE $CLUSTER_GENERATED_FILE"
-echo "You should now manually change your cluster configuration by editing the generated files."
+cat "$MASTER_USER_DATA" \
+    | sed -e "s#\$OPENSTACK_CLOUD_PROVIDER_CONF#$OPENSTACK_CLOUD_PROVIDER_CONF#" \
+    > $USERDATA/$PROVIDER_OS/master-user-data.sh
+cat "$WORKER_USER_DATA" \
+  | sed -e "s#\$OPENSTACK_CLOUD_PROVIDER_CONF#$OPENSTACK_CLOUD_PROVIDER_CONF#" \
+  > $USERDATA/$PROVIDER_OS/worker-user-data.sh
+
+echo $CLOUD > $CONFIG_DIR/os_cloud.txt
```

Comment:
Could you please use `echo -n "$CLOUD"` here.. And maybe even  `echo -n "$CLOUD" | tr -d '\n'`, I guess you know why ;)

Reply:
how about printf instead